File: VB/Scaffolding.DetailCollections/DepartmentContextDataModel/Model/IDepartmentContextUnitOfWork.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.DetailCollections.Common.Utils
Imports Scaffolding.DetailCollections.Common.DataModel
Imports Scaffolding.DetailCollections.Common.DataModel.EntityFramework
Imports Scaffolding.DetailCollections.Model

Namespace Scaffolding.DetailCollections.DepartmentContextDataModel
	Public Interface IDepartmentContextUnitOfWork
	Inherits IUnitOfWork
		Function HasChanges() As Boolean
		ReadOnly Property Courses() As ICourseRepository
		ReadOnly Property Departments() As IDepartmentRepository
		ReadOnly Property Employees() As IEmployeeRepository
	End Interface
End Namespace

File: CS/Scaffolding.DetailCollections/DepartmentContextDataModel/Model/IDepartmentContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.DetailCollections.Common.Utils;
using Scaffolding.DetailCollections.Common.DataModel;
using Scaffolding.DetailCollections.Common.DataModel.EntityFramework;
using Scaffolding.DetailCollections.Model;

namespace Scaffolding.DetailCollections.DepartmentContextDataModel {
    public interface IDepartmentContextUnitOfWork : IUnitOfWork {
        bool HasChanges();
        ICourseRepository Courses { get; }
        IDepartmentRepository Departments { get; }
        IEmployeeRepository Employees { get; }
    }
}
